opensrp-opd/src/main/java/org/smartregister/opd/utils/JsonFormUtils.java

```diff
@@ -0,0 +1,1031 @@
+            List<Object> vall = new ArrayList<>();
+            vall.add(referenceEvent.getProviderId());
+            event.addObs(new Obs(FORM_SUBMISSION_FIELD, DATA_TYPE, formSubmissionField, "", vall, new ArrayList<>(), null,
+                    formSubmissionField));
+
+            formSubmissionField = "From_LocationId";
+            vall = new ArrayList<>();
+            vall.add(fromLocationId);
+            event.addObs(new Obs(FORM_SUBMISSION_FIELD, DATA_TYPE, formSubmissionField, "", vall, new ArrayList<>(), null,
+                    formSubmissionField));
+
+            formSubmissionField = "To_ProviderId";
+            vall = new ArrayList<>();
+            vall.add(toProviderId);
+            event.addObs(new Obs(FORM_SUBMISSION_FIELD, DATA_TYPE, formSubmissionField, "", vall, new ArrayList<>(), null,
+                    formSubmissionField));
+
+            formSubmissionField = "To_LocationId";
+            vall = new ArrayList<>();
+            vall.add(toLocationId);
+            event.addObs(new Obs(FORM_SUBMISSION_FIELD, DATA_TYPE, formSubmissionField, "", vall, new ArrayList<>(), null,
+                    formSubmissionField));
+
+            addMetaData(context, event, new Date());
+
+            JsonFormUtils.tagSyncMetadata(event);
+            return event;
+
+        } catch (Exception e) {
+            Timber.e(e, "JsonFormUtils --> createMoveToCatchmentEvent");
+            return null;
+        }
+    }
+
+
+    public static OpdEventClient processOpdDetailsForm(String jsonString,FormTag formTag) {
+        try {
+            Triple<Boolean, JSONObject, JSONArray> registrationFormParams = validateParameters(jsonString);
+
+            if (!registrationFormParams.getLeft()) {
+                return null;
+            }
+
+            JSONObject jsonForm = registrationFormParams.getMiddle();
+            JSONArray fields = registrationFormParams.getRight();
+
+            String entityId = getString(jsonForm, ENTITY_ID);
+            if (StringUtils.isBlank(entityId)) {
+                entityId = generateRandomUUIDString();
+            }
+
+            processGender(fields);//multi language to re visit
+
+            processLocationFields(fields);
+
+            lastInteractedWith(fields);
+
+            dobUnknownUpdateFromAge(fields);
+
+            processReminder(fields);
+
+            Client baseClient = org.smartregister.util.JsonFormUtils.createBaseClient(fields, formTag, entityId);
+
+            Event baseEvent = org.smartregister.util.JsonFormUtils
+                    .createEvent(fields, getJSONObject(jsonForm, METADATA), formTag, entityId,
+                            Utils.metadata().getRegisterEventType(), Utils.metadata().getTableName());
+
+            JsonFormUtils.tagSyncMetadata(baseEvent);// tag docs
+
+            return new OpdEventClient(baseClient, baseEvent);
+        } catch (Exception e) {
+            Timber.e(e, "JsonFormUtils --> processDetailsForm");
```

Comment:
Can we catch a more specific exception?

Reply:
no need since the whichever exception it throws will extend `Exception` so the class name will be given e.g `java.lang.IllegalArgumentException` , i think we usually specify a specific exception when we want do another thing when its thrown e.g either retry or something

Reply:
Yea, I think Exception covers everything. Right? Do we want to catch everything. Even `RuntimeException` is a child of `Exception`?

It also helps with readability. We expect X, Y, Z exceptions to happen and are OK handling them in this method. Anything else should be handled by the method calling this so they know something wrong happened